File: problem48.py
const = 10000000000
def add(a, b):
	return (a + b) % const

def mul(a, b):
	return (a * b) % const

def pow(n, m):
	s = 1
	for i in range(m):
		s = mul(s, n)
	return s

s = 0
for i in range(1, 1001):
	s = add(s, pow(i, i))
print s

File: problem48.hs
con = 10000000000
add a b = rem (a + b) con
mul a b = rem (a * b) con
pow a 1 = mul a 1
pow a n = mul a (pow a (n - 1))
cal = sum . map (\x -> (pow x x))
main = print(rem (cal [1..1000]) con)